apps/examples/testcase/le_tc/filesystem/tc_fs_smart_sfs.c

```diff
@@ -0,0 +1,897 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file sfs.c
+/// @brief Test Case Example for Secure File System Feature
+
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+
+#include <stdio.h>
+#include <tinyara/config.h>
+#include <tinyara/sched.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include "tc_internal.h"
+#include "../../tc_common.h"
+
+#define MAXBUFSIZE 256
+#define CRYPTSIZE 16
+#define SFS_TASK_STACKSIZE 8192
+#define SFS_thread_name "SFS_TC"
+
+char testdata[MAXBUFSIZE];
+char data[MAXBUFSIZE] = "";
+static const char *FILE_PATH = "/mnt/name.txt";
+
+/**
+* @fn                   :file_writetest_fwrite
+* @brief                :writing into the specified file using fwrite function
+* @return               :void
+*/
+
+static void file_writetest_fwrite(char *dat, size_t len)
+{
+	FILE *fp;
+	/*If file already exists, delete the file */
+
+	remove(FILE_PATH);
+	fp = fopen(FILE_PATH, "w+");
+	if (fp != NULL) {
+		fwrite(dat, 1, len, fp);
+	}
+	fclose(fp);
+}
+
+/**
+* @fn                   :file_writetest_fread
+* @brief                :reading the specified file using fread function
+* @return               :void
+*/
+
+static void file_readtest_fread(char *dat)
+{
+	FILE *fp;
+	size_t len;
+
+	fp = fopen(FILE_PATH, "r+");
+	fseek(fp, 0, SEEK_END);
+	len = ftell(fp);
+	fseek(fp, 0, SEEK_SET);
+	if (fp != NULL) {
+		fread(dat, 1, len, fp);
+	}
+	fclose(fp);
```

Comment:
1. Is it ok to check (fp != NULL) after fseek?
2. same as previous. close should be inside {   }
3. same for belows